### PR TITLE
Use "sha256sum" on Windows 

### PR DESCRIPTION
**Pull request by sschuberth**
Some changes around using "sha256sum" instead of "shasum" on Windows, in particular on MSYS2.

### CODE REVIEW

File: test/testhelpers.sh [hunk 1]
```diff
@@ -428,7 +428,19 @@ comparison_to_operator() {
...
 calc_oid() {
-  printf "$1" | shasum -a 256 | cut -f 1 -d " "
+  if [ $IS_WINDOWS -eq 1 ]; then
+    printf "$1" | sha256sum
+  else
+    printf "$1" | shasum -a 256
+  fi | cut -f 1 -d " "
+}
+
+calc_oid_file() {
```
Comment (ttaylorr) [inline test/testhelpers.sh]: I know that there wasn't documentation on `calc_oid` before, but adding documentation to both of these functions would be incredibly helpful 👍

Comment (sschuberth) [inline test/testhelpers.sh]: Sure, but could you be a bit more clear on what to document? I mean, should I basically just use the commit message and document *why* to use sha256sum on Windows, or do you want to have the pipe of commands itself documented?

Comment (ttaylorr) [inline test/testhelpers.sh]: Sure, let's just document what these functions take in their argument and that they work on both macOS and Windows.

Comment (sschuberth) [inline test/testhelpers.sh]: Like so?

Comment (ttaylorr) [inline test/testhelpers.sh]: Perfect!

File: test/testhelpers.sh [hunk 2]
```diff
@@ -427,8 +427,28 @@ comparison_to_operator() {
...
+# Calculate the object ID from the file passed as the argument
+calc_oid_file() {
+  # Windows might be MSYS2 which does not have the shasum Perl wrapper
+  # script by default, so use sha256sum directly. MacOS on the other hand
+  # does not have sha256sum, so still use shasum otherwise.
+  if [ $IS_WINDOWS -eq 1 ]; then
+    sha256sum "$1"
+  else
+    shasum -a 256 "$1"
+  fi | cut -f 1 -d " "
```
Comment (larsxschneider) [inline test/testhelpers.sh]: I guess there is no reasonable nice way to have the decision between `sha256sum` and `shasum` once in the code? Maybe a `sha256` function? I guess "eval" is too evil?!

**Comment by larsxschneider** [comment]
@sschuberth The unit tests run clean on my Win7 box but the integration tests don't work. Should they?
```
$ ./script/integration
/c/path/to/my/workspace/src/github.com/github/git-lfs/test/remote/url still exists!
Confirm other tests are done, and run:
  $ curl http://127.0.0.1:51043/shutdown
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 127.0.0.1 port 51043: Connection refused
```

Plus, I can't find any output of the unit or integration tests in the appveyor logs. Should they be there? Maybe I am not looking at the right places - I am not familiar with appveyor.

Thanks for working on the Windows CI check! 👍

**Comment by technoweenie** [comment]
@larsxschneider: that's an indication of a previous unclean test run. Just delete `/c/path/to/my/workspace/src/github.com/github/git-lfs/test/remote/url` and try it again.

**Comment by sschuberth** [comment]
> The unit tests run clean on my Win7 box but the integration tests don't work. Should they?

Yes, and they do on my machine [as well as on CI](https://ci.appveyor.com/project/sschuberth/git-lfs/build/1.0.92).

> Plus, I can't find any output of the unit or integration tests in the appveyor logs. Should they be there?

Yes, see the same link as above. However, the test results do not show up on the [tests tab](https://ci.appveyor.com/project/sschuberth/git-lfs/build/1.0.92/tests) yet as AppVeyor only supports [a limited set of test result formats](https://www.appveyor.com/docs/running-tests/#pushing-real-time-test-results-to-build-console). We would first need to write some simple conversion script, but I wanted to leave that for later.

**Comment by larsxschneider** [comment]
Of course... the URL file. I should have read the error properly. Sorry.

All but two integration tests pass on my local Windows box:
```
test: attempt private access without credential helper ...         FAILED
test: push --all (no ref args) ...                                 FAILED
```

However, that's probably related to some of my setup. If they pass on the CI that this is good enough I think.

LGTM 👍

**Comment by ttaylorr** [comment]
Let's do it!

**Pull request merged**